Add mediator tab and add mediator screen

craigsailor opened the pull request:
I added some tabs to the arbitrator UI for the future mediation requests and mediation messages.
Also, I added the "add Mediator" screen and in the core added the "disputes" part that is mostly copied from the "arbitration" but will be changed a piece at a time to eliminate the arbitration.

rpgoldberg committed:
List Qbase (QBS)

freimair committed:
Use same private_key in NewTor and RunningTor mode

Commit:
Update Horizen (ZEN)

Commit:
added mediation tabs

Commit:
working additional tabs

Commit:
working tabs

aistrych committed:
List Pinkcoin (PINK)

camthegeek committed:
List Aeon (AEON)

freimair committed:
Basic framework for Monitor

freimair committed:
Basic configuration infrastructure available

freimair committed:
Periodical schedule

freimair committed:
Use configurable properties file

freimair committed:
Reloading config during runtime available

freimair committed:
Metrics can be disabled via config

freimair committed:
Basic configuration tests

freimair committed:
Metrics can be reenabled via config

freimair committed:
Metrics can be renamed

freimair committed:
Use lombok/slf4j for logging

freimair committed:
Configure gradle properly

System96 committed:
List FourtyTwo (FRTY)

Official project URL: [http://coin42.co] (Temporary address: http://coin42.tk)
Official block explorer URL: [http://explorer.coin42.co]

devopsralf committed:
List Remix (RMX)

georgemcarlson committed:
List SiaPrimeCoin (SCP)

Commit:
Added Mediator Registration

Commit:
added CMD-M shortcut to open add mediator

Commit:
removed get arbitrator section

Commit:
added mediation language warning

freimair committed:
Introduce general purpose Tor instance

freimair committed:
Synchronize all the things

freimair committed:
Introduce reporting stub

freimair committed:
Add initial TorStartupTime metric

freimair committed:
Enhance config file documentation

freimair committed:
Cleanup

freimair committed:
Configurable socksPort for TorStartupTime Metric

freimair committed:
TorHiddenServiceStartupTime Metric available

freimair committed:
TorRoundTripTime Metric available

oscarguindzberg committed:
Add missing @Override

oscarguindzberg committed:
BisqWalletFactory: Remove method already declared on superinterface

oscarguindzberg committed:
BisqRiskAnalysis: update comment

arunasurya committed:
Correct minor grammatical mistakes in the T&C

freimair committed:
Refactored Metrics scheduler

ManfredKarrer committed:
Merge pull request #1948 from Qbase-Foundation/list-qbase-asset

List Qbase (QBS)

ManfredKarrer committed:
Merge pull request #2004 from tarrenj/List_Horizen

List Horizen (ZEN)

ManfredKarrer committed:
Merge branch 'master' into add_mediator_tab

# Conflicts:
#	desktop/src/main/java/bisq/desktop/main/account/AccountView.java

ManfredKarrer committed:
Merge pull request #2116 from a3rk/list-remix

List Remix (RMX)

ManfredKarrer reviewed (changes requested):
When I register a mediator I get a nullpointer:

java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at bisq.core.disputes.MediatorManager.updateMediatorMap(MediatorManager.java:226)

It is because the onAllServicesInitialized of MediatorManager is not called. Need to be called same like arbitr. Manager. 

The M is reserved alreayd for another features, maybe use Y which is not used so far for opening the registration

I merged master and will make a PR.

steevebrush committed:
List Iridium (IRD)

ManfredKarrer committed:
Add FIXME

ManfredKarrer committed:
Use R instead of M for opening mediator tab, use Y for arbitrator

- use if else instead of new handler
- Make addButtons method protected

ripcurlx committed:
Merge pull request #2138 from arunasurya/useragre

Correct minor grammatical mistakes in the T&C

ripcurlx committed:
Merge pull request #2137 from oscarguindzberg/codecleanup1

Code clean up

ManfredKarrer commented:
I made a PR with merge and small changes. Still need the fix with the onAllServicesInitialized call. At the first popup there is still the old text with arbitration as well.

ManfredKarrer committed:
Add @Override

freimair committed:
TorRoundtripTime Metric does samples

ripcurlx committed:
Remove unnecessary backslash

ManfredKarrer committed:
Merge pull request #2102 from camthegeek/list-aeon

List Aeon (AEON)

ManfredKarrer committed:
Merge pull request #2148 from ripcurlx/fix-minor-translation-typo

Remove unnecessary backslash

ManfredKarrer committed:
Merge pull request #2144 from iridiumdev/List_Iridium(IRD)

List Iridium (IRD)

oscarguindzberg committed:
Fix getActiveKeyChain() typo

ripcurlx committed:
Merge pull request #2128 from geo-gs/list-siaprimecoin

List SiaPrimeCoin (SCP)

ripcurlx committed:
Merge pull request #2097 from Astrych/list-pinkcoin

List Pinkcoin (PINK)

ripcurlx committed:
Merge pull request #2151 from oscarguindzberg/getActiveKeychain-typo

Fix getActiveKeyChain() typo

craigsailor committed:
Merge branch 'master' into add_mediator_tab

craigsailor reviewed (commented):
I'll fix the onAllServicesInitialized thing in the next pull request. I found where that's supposed to be in the BisqSetup.java file.

tlightsky committed:
List SpaceCash (SPACE)

freimair committed:
Graceful shutdown, Netlayer 0.6.2

freimair committed:
Refactored TorStartupTime

Commit:
removed extra addbutttons

freimair committed:
Update Netlayer to 0.6.2 and thus, Tor to 8.0.4

CaveSpectre11 committed:
List UnitedCommunityCoin (UCC)

CaveSpectre11 committed:
List Dextro (DXO)

CaveSpectre11 committed:
List IdaPay (IDA)

CaveSpectre11 committed:
List GambleCoin (GMCN)

CaveSpectre11 committed:
Fix typos and improve grammer

craigsailor committed:
Merge branch 'add_mediator_tab' into mediator-tab

craigsailor committed:
Merge pull request #6 from ManfredKarrer/mediator-tab

Mediator tab

devinbileck committed:
Update windows install experience

- Increase the minimum supported OS version to Windows 7 with SP1.
Previously it was Windows XP which has been EOL for some time now.
Hopefully no one is still using Windows XP.

- Show a final dialog once the install has completed. This will inform
the user that the install has completed with an option to launch the
application. Previously it would skip this final dialog and immediately
launch the application, but the user may not want to launch it
immediately.

- Create an application shortcut on the user's desktop.

- Check if the application is running prior to install or uninstall and
abort with a prompt to the user that the application must first be
closed.

- German, French, and Spanish translations. The installer should
auto-detect the OS language and display all text in the appropriate
language. For now I have just added a couple languages, but it is very
easy to add any of the other languages supported by Inno Setup.

- Additionally, I removed a few unnecessary items that weren't being
utilized such as installing a service.

devinbileck committed:
Update install_java.bat

Do not download 7zip if it had previously been downloaded, for example
if this script had previously been run. And don't delete it at the end
of this script in case this script is run again. This is an attempt to
save time and bandwidth.

ManfredKarrer committed:
Fix mediator instantiation

ManfredKarrer committed:
Merge remote-tracking branch 'origin/mediator-tab' into mediator-tab

ManfredKarrer committed:
Merge pull request #2162 from CaveSpectre11/text-edits

Fix typos and improve grammar

ManfredKarrer committed:
Merge pull request #1970 from HyperspaceApp/list-spacecash

List SpaceCash (XSC)

ManfredKarrer committed:
Merge pull request #2150 from FourtyTwo/master

List FourtyTwo (FRTY)

ManfredKarrer committed:
Merge pull request #2157 from CaveSpectre11/list-dextro-asset

List Dextro (DXO)

ManfredKarrer committed:
Merge pull request #2158 from CaveSpectre11/list-idapay-asset

List IdaPay (IDA)

craigsailor committed:
Merge pull request #8 from ManfredKarrer/mediator-tab

Fix Mediator registration

ManfredKarrer committed:
Merge pull request #2159 from CaveSpectre11/list-ucc-asset

List UnitedCommunityCoin (UCC)

CaveSpectre11 committed:
Merge branch 'master' into list-gamblecoin-asset

sqrrm committed:
Fix icon handling in vote result table, fix #2141

freimair committed:
Introduce reporter concept

freimair committed:
Refactored Metric

freimair committed:
Refactored metric infrastructure

freimair committed:
Graphite reporter is available

freimair committed:
Pretty up config file

oscarguindzberg committed:
Remove Networks and AbstractBitcoinNetParams since they are not used

ManfredKarrer committed:
Merge pull request #2172 from sqrrm/dao-scrolling

Fix icon handling in vote result table, fix #2141

ManfredKarrer committed:
Merge pull request #2182 from oscarguindzberg/remove-unused-classes

Remove Networks and AbstractBitcoinNetParams since they are not used

freimair committed:
Readme, Javadoc, cleanup

freimair committed:
Dependencies and Buildfile cleanup

freimair committed:
Fixed compiation error as in: Commit of shame

freimair committed:
Removed unused dummy Metric

ManfredKarrer committed:
Reformat code, organize imports, fix spelling

ManfredKarrer committed:
Rename TorRoundtripTime to TorRoundTripTime

ManfredKarrer committed:
Add monitor directory as parent for test directories

- Apply reorg imports and reformat to test classes

ManfredKarrer committed:
Update gitignore

- Add /monitor/TorHiddenServiceStartupTimeTests/* and
/monitor/monitor-tor/*

ManfredKarrer committed:
Apply code inspection

- Fix spelling
- Remove exception throw clause which was never thrown
- Add not null preconditions
- Remove unneeded types in diamonds
- Add final
- Fix java doc
- Make Reporter constructor protected (abstract)

ManfredKarrer committed:
Add guava

ManfredKarrer committed:
Use 1.1.10 instead of $logbackVersion

$logbackVersion causes build errors.

ManfredKarrer committed:
Update hashes

ManfredKarrer committed:
Ignore tests

- Revert spelling error as misspelling was used for test
- Add TODO for failing test (timing issue, seems to fail just sometimes)
- Set @Disable to test classes which take longer as Tor is started

ManfredKarrer committed:
Merge pull request #2009 from freimair/use-same-address-for-native-and-external-tor

Use same private_key in NewTor and RunningTor mode + Hiddenservice Fix

freimair committed:
Merge branch 'master' into monitor

freimair committed:
Merge branch 'monitor-nits' of git://github.com/ManfredKarrer/bisq into ManfredKarrer-monitor-nits

freimair committed:
Merge branch 'ManfredKarrer-monitor-nits' into monitor

ManfredKarrer committed:
Merge pull request #2181 from freimair/monitor

Bisq Network Monitor: Babysteps

BravoIndia committed:
Update displayStrings.properties

Clarifying/correcting grammar of bug report message.

ManfredKarrer committed:
Merge pull request #2185 from BravoIndia/patch-3

Update displayStrings.properties (error & bug report message)

ManfredKarrer committed:
Merge pull request #2161 from CaveSpectre11/list-gamblecoin-asset

List GambleCoin (GMCN)

ManfredKarrer committed:
Add toString methods

ManfredKarrer committed:
Improve logging

ManfredKarrer committed:
Comment out debug logs

ManfredKarrer committed:
Add comment

ManfredKarrer committed:
Add peer address to error log

- Add full toString method to Connection.SharedModel

ManfredKarrer committed:
Only log warning if mainnet

Commit:
added updates

ManfredKarrer committed:
Add more logs

ManfredKarrer committed:
Improve error logs

ManfredKarrer committed:
Add number of blocks in comment

ManfredKarrer committed:
Remove final qualifier

ManfredKarrer committed:
Use lambda

ManfredKarrer committed:
Fix handling of case when no seed node is available.

In regtest we got incorrectly onNoPeersAvailable called if the first
seed node in our list was not available. It triggered a
onAllServicesInitialized and led to incorrect behaviour in case we
received data from the second seednode. One bug was that we did not
clone the remaining nodes list in the for loop which led that the list
was empty for each delayed call. The other bug was that we did not
check if there is another connection attempt open when we got a
connection fault from the not available seed.

ManfredKarrer committed:
Change handling of addListeners and start calls

We add first all listeners at all services and then call the start methods.
Some services are listening on others so we need to make sure that the
listeners are set before we call start as that might trigger state change
which triggers listeners.

- Change order of ballotListService and proposalService to reflect
dependency.

ManfredKarrer committed:
Fix startup behaviour.

Call fillListFromProtectedStore and fillListFromAppendOnlyDataStore
at start. The listeners are not triggered as registered after we have
received the initial data. So we need to read the P2P network data at
the start method.

ManfredKarrer committed:
Improve logging

ManfredKarrer committed:
Revert to netlayer 0.6 because of an issue with seednodes

If a seednode is using netwlayer version 0.6.2 (or 0.6.1) it is not
reachable from other nodes. Still unclear what causes that and if it
would affect also normal Bisq nodes, but to be careful we revert to
latest stable netlayer version which fixes the issue.

ManfredKarrer committed:
Merge pull request #2192 from ManfredKarrer/revert-netlayer-update

Revert to netlayer 0.6 because of an issue with seednodes

ManfredKarrer committed:
Improve logging

ManfredKarrer committed:
Add chainTipHeight property

- We want to know for the vote reveal service what is the latest block.
Currently that is not exposed in the DAO only in the BitcoinJ classes,
but we don't want to access it from there.

ManfredKarrer committed:
Use chain tip height for vote reveal tx publishing

- Use the chain tip height and not the current chain height in the
parsing to check if we are in the correct phase and cycle. Only
publish the tx if we are in the correct cycle.

freimair committed:
Introduce netlayer release 0.6.3

ManfredKarrer committed:
Handle late vote reveal txs

If the phase and cycle for the vote reveal tx was missed we still
publish it but it is considered invalid. We do not throw an exception
but filter such txs away from the vote result evaluation.

We cannot use the strategy to unlock the BSQ from the vote tx in such a
case because the blind vote tx is already in the past and is not parsed
again (snapshot).
Alternatively we could have used a different tx type for the unlock
purpose but we prefer to keep such an exceptional case simple.

ManfredKarrer committed:
Merge pull request #2194 from freimair/netlayer063

Introduce netlayer release 0.6.3

ManfredKarrer committed:
Merge branch 'master' into dao-fix-late-proposals-processing

ManfredKarrer committed:
Merge branch 'master' into dao-fix-vote-reveal-tx-publishing

oscarguindzberg committed:
Remove "pricenode" direct dependency on "assets"

It depends on "assets" already because it is a depencendy of "core"

ManfredKarrer committed:
Merge pull request #2197 from oscarguindzberg/asset-dependency

Remove "pricenode" direct dependency on "assets"

oscarguindzberg committed:
Be able to upgrade bitcoinj without upgrading libdohj

ManfredKarrer committed:
Set chainTipHeight before parsing.

ManfredKarrer committed:
Refactor maybeRevealVotes method.

Add exception handling to revealVote

ManfredKarrer committed:
Remove phase and cycle check for vote reveal txs.

We do not check phase or cycle as a late voteReveal tx is considered
a valid BSQ tx. The vote result though will ignore such votes.

Add more log info.

ManfredKarrer committed:
Ignore items if isBlindVoteInCorrectPhaseAndCycle is false.

If we get a voteReveal tx which was published too late we ignore it.
- Refactorings
- Improve logs

ManfredKarrer committed:
Refactorings

ManfredKarrer committed:
Improve logs

ManfredKarrer committed:
Improve logs

ManfredKarrer committed:
Move logs and comments to 'if' clause

ManfredKarrer committed:
Change log level

ManfredKarrer committed:
Call onParseBlockComplete listeners only after onParseBlockChainComplete

To avoid performance issues we omit the listener calls at
onParseBlockComplete as long we are in the batch processing of blocks.
Without that optimization it took about 50 sec. for 4000 blocks, with it
we could reduce it to 4 seconds.

ManfredKarrer committed:
Add duration for batch parsing to logs

ManfredKarrer committed:
Add db storage files

ripcurlx committed:
Merge pull request #2164 from devinbileck/update_windows_install_experience

Update windows install experience

ripcurlx committed:
Merge pull request #2165 from devinbileck/update-install_java.bat

Update install_java.bat

ripcurlx committed:
Merge pull request #2198 from oscarguindzberg/freebitcoinj

Be able to upgrade bitcoinj without upgrading libdohj

Commit:
put back arbitrator unlock button since we still need it

ManfredKarrer committed:
Merge pull request #2191 from ManfredKarrer/dao-fix-late-proposals-processing

Dao: fix late proposals processing

ManfredKarrer committed:
Adjust % value for offer price tolerance window

- There have been some reports that the 0,5% value is too narrow so we
increase it to 1%.

ManfredKarrer committed:
Merge branch 'master' into dao-fix-vote-reveal-tx-publishing

ManfredKarrer committed:
Merge pull request #2195 from ManfredKarrer/dao-fix-vote-reveal-tx-publishing

Dao: fix vote reveal tx publishing

ManfredKarrer committed:
Merge branch 'master' into dao-fix-performance-issue-at-parsing

ManfredKarrer committed:
Merge pull request #2199 from ManfredKarrer/dao-fix-performance-issue-at-parsing

Dao: Fix performance issue at parsing

ManfredKarrer committed:
Add comment

ManfredKarrer committed:
Remove duplicated ArbitratorManager

ManfredKarrer committed:
Merge branch 'master' into mediator-tab

ManfredKarrer committed:
Add comment

ManfredKarrer committed:
Fix typo

ManfredKarrer committed:
Fix package

ManfredKarrer commented:
I added a few small fixes and update to master on https://github.com/craigsailor/bisq/pull/13. If that gets in that PR is ACKed.

ManfredKarrer commented:
Would be good to rebase as well to get a cleaner Git history...

craigsailor committed:
Merge pull request #13 from ManfredKarrer/mediator-tab

Mediator tab
